### ISSUE TITLE
"internet conection is not active"

### ISSUE THREAD

**Issue by Lithium64** [teejee2008/ukuu#49]
Since the last update, mine Ukuu only shows this message and doesn't show any kernel to download.
![captura de tela de 2018-01-18 14-53-56](https://user-images.githubusercontent.com/10705488/35113305-b65eea88-fc5f-11e7-9b7e-e89c54f0cedb.png)

Ubuntu 17.10
Kernel 4.14.13

**Comment by kartun83** [comment]
Same for me on linux mint 18.3

**Comment by siddht4** [comment]
try to update your ukuu version from sources.  I had faced this issue months back and its fixed now.  You can check my issue too.https://github.com/teejee2008/ukuu/issues/19. This https://github.com/teejee2008/ukuu/commit/219ca76f413d4e4c490c8ade5170fde1cb268989    pull request fixed the issue for me. So try to update it

**Comment by siddht4** [comment]
@Lithium64  What`s your ukuu version by the way?

**Comment by Lithium64** [comment]
@siddht4 
ukuu v18.1
Distribution: Ubuntu 17.10
Architecture: amd64
Running kernel: 4.14.13-041413-generic
Kernel version: 4.14.13-041413-generic

I'm runing the latest build from ppa:teejee2008/ppa

**Comment by kartun83** [comment]
@siddht4 
ukuu --version
ukuu v18.1
Distribution: Linux Mint 18.3 Sylvia
Architecture: amd64
Running kernel: 4.14.11-041411-generic
Kernel version: 4.14.11-041411-generic
E: Unknown option: --version
E: Run 'ukuu --help' to list all options

**Comment by siddht4** [comment]
looks like my prevous issue has re-surfaced. 

@Lithium64 @kartun83    this is the bug,I had issues with and looks like it has resurfaced again.But https://github.com/teejee2008/ukuu/commit/219ca76f413d4e4c490c8ade5170fde1cb268989   has actually fixed it as far as I know.

**Comment by Lithium64** [comment]
@siddht4 From what I could see there were changes in the connection code after version 17.12.1, which caused this regression.

**Comment by Nevel** [comment]
For what it's worth, I can reproduce this issue as well.

ukuu v18.1
Distribution: Linux Mint 18.3 Sylvia
Architecture: amd64
Running kernel: 4.10.0-42-generic
Kernel version: 4.10.0.42.46

**Comment by ams001** [comment]
Likewise with Nevel.  Oddly, it also reports an updated kernel, followed by:
*No Internet*
Internet connection is no Active

**Comment by tddts** [comment]
Same here.

Ukuu 18.1
Distribution: Linux Mint 18.3 Sylvia
Architecture: amd64
Kernel: 4.13.0-32

**Comment by tddts** [comment]
Also a very strange thing happened: applications started to close very slow and after a while desktop got frozen and HDD LED was showing constant work. Not sure if it happened because of Ukuu, because I closed it a while ago. And after reboot I got two dialogs popping up and a notification. Both first dialog and a notification were something like "New version 4.15 is available" and second one was "Internet connection is not active".

**Comment by Nevel** [comment]
Yes, I also get the notification that a new version has been released, prior to the error message stating that the connection is not active.

**Comment by kartun83** [comment]
Is it using different network code for checks and pulling kernels ?

**Comment by CottonEaster** [comment]
I had this problem, after deactivating my VPN. Ukuu returned to normal after a reboot.

Mint 18.3.

```
ukuu --version
ukuu v18.1
E: Error creating directory: Permission denied
E: Failed to create dir: /tmp/ukuu/rqoIWfyy
Distribution: Linux Mint 18.3 Sylvia
Architecture: amd64
Running kernel: 4.15.4-041504-generic
Kernel version: 4.15.4-041504-generic
E: Unknown option: --version
E: Run 'ukuu --help' to list all options
```

**Comment by cmera** [comment]
I emptied /tmp/ukuu and /tmp/ukuu-gtk and it seemed to stop complaining..

**Comment by Nevel** [comment]
Thanks for reporting, that worked for me as well.

**Comment by ams001** [comment]
Not me, but `sudo apt-get remove ukuu` did :-(

**Comment by zkcode** [comment]
it doesn't work for me also

ukuu v18.1
Distribution: Ubuntu 16.04.4 LTS
Architecture: amd64
Running kernel: 4.13.16-041316-generic
Kernel version: 4.13.16-041316-generic

**Comment by HougeLangley** [comment]
Same error. Using proxy could fix this. By the way, Do not using ukuu-gtk

**Comment by dreamcat4** [comment]
Hello. This error is being caused by slow / unresponsive DNS lookup requests. It seems that the program `ukuu` has some type of short built-in timeout. Where if there is no response, then it gives up too soon for the system to finish processing the request.

On my system, the command `ping` would always take at least 5 seconds because the system dns resolver would wait 5 seconds before returning the DNS lookup result. So (for me), it was solved by taking some of the following suggestions:

http://www.math.tamu.edu/~comech/tools/linux-slow-dns-lookup/

Specifically by appending the line: `options single-request` to my `/etc/resolv.conf` file. (it does keep getting overwritten though, so there is a different place in the system config files to add it to, somewhere else).

Trying the 

```
GRUB_CMDLINE_LINUX_DEFAULT="ipv6.disable=1 quiet"
```

in your `/etc/default/grub` file might also help. Not sure. Anyway, this seems to be a common problem no on new versions of ubuntu, such as 18.04.


Furthermore:

Now that the cause of this issue has been found. I would still argue that exposing some user configuration mechanism to override the default internal timeout to `ukuu` program is still desirable. Because there are a few users who must connect via satellite endpoints. (and for those users, the network latency can be very high indeed). Which admittedly is an edge case.

However adding the configuration option should not be too hard, and it would also be helpful for any other users experiencing connectivity issues. For example - inside a corporate LAN, which the user cannot do anything about. Or other networking issues where they solution simply is not clear. Sometimes, it's just a much bigger priority for these people, update their kernels first, before looking at such other problems!

Have added this same comment to issue #76 also
Kind regards

**Comment by iandol** [comment]
Same problem, and I trying the DNS fixes above do not solve it for me (edited /etc/resolv.conf appending `options single-request` and inactivating ipv6 completely, then flushing the DNS cache for good measure.

```
❯ sudo ukuu --check
ukuu v18.5.1
Distribution: Ubuntu 18.04.1 LTS
Architecture: amd64
Running kernel: 4.18.0-041800-lowlatency
Kernel version: 4.18.0-041800-lowlatency
Cache: /home/aristotle/.cache/ukuu
Temp: /tmp/ukuu/8Lsgt7uK
E: Internet connection is not active
E: Internet connection is not active
----------------------------------------------------------------------
Found installed: 4.13.0-46.51
Found installed: 4.15.0-32.35
Found installed: 4.15.0.32.34
Found installed: 4.18.0-041800.201808122131
----------------------------------------------------------------------
Latest point update: 4.18.4
----------------------------------------------------------------------

~ 12s

❯ ping kernel.ubuntu.com
PING kernel.ubuntu.com (91.189.94.216) 56(84) bytes of data.
64 bytes from wani.canonical.com (91.189.94.216): icmp_seq=1 ttl=48 time=338 ms
64 bytes from wani.canonical.com (91.189.94.216): icmp_seq=2 ttl=48 time=339 ms
64 bytes from wani.canonical.com (91.189.94.216): icmp_seq=3 ttl=48 time=338 ms
64 bytes from wani.canonical.com (91.189.94.216): icmp_seq=4 ttl=48 time=338 ms
64 bytes from wani.canonical.com (91.189.94.216): icmp_seq=5 ttl=48 time=338 ms

--- kernel.ubuntu.com ping statistics ---
5 packets transmitted, 5 received, 0% packet loss, time 4980ms
rtt min/avg/max/mdev = 338.053/338.570/339.238/0.655 ms
```

**Comment by dreamcat4** [comment]
@iandol it's not your `ping <anything>` result that matters. It is the `time ping <anything>`. You need to put the `time` command in front to see the delay which is due to your own system's DNS resolver.

**Comment by CottonEaster** [comment]
I have the problem on Mint 19, and it seems that in my case the problem owes to a momentary VPN dropout. It seems to me that _if Ukuu could wait a little longer before declaring there is no Internet_, then, at least in my case, that would solve the problem.

Also, #76 seems a duplicate of this one.

**Comment by iandol** [comment]
Hi @dreamcat4 thanks for the reply. My time to respond is :

```
❯ time ping -c 1 kernel.ubuntu.com
PING kernel.ubuntu.com (91.189.94.216) 56(84) bytes of data.
64 bytes from wani.canonical.com (91.189.94.216): icmp_seq=1 ttl=49 time=359 ms

--- kernel.ubuntu.com ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 359.949/359.949/359.949/0.000 ms

ping -c 1 kernel.ubuntu.com  0.00s user 0.00s system 0% cpu 0.364 total
```
I am in China, and we do have quite a bit of fluctuation in our connectivity but even when stable and not too slow, ukuu never connects...

**Comment by ams001** [comment]
I don't believe its down to the resolver because I use a caching nameserver, so while the first attempt may fail as the cache is loaded, the second should work as the resolver returns immediately.  It appears to be the connection speed to wani.canonical.com since failure was more common when I was on the road (and slower access e.g. pings around 300ms+) than on our corporate network (e.g. pings around 21ms).

**Comment by iandol** [comment]
For those with this problem, this bash script is working fine: https://github.com/pimlie/ubuntu-mainline-kernel.sh

**Comment by MrFunken** [comment]
Is there any way to disable this useless connection check?

**Comment by NathanIceSea** [comment]
This problem is really annoying, hopefully adding a parameter to disable connection verifying.

**Comment by Pipeliner** [comment]
Still the same problem with ukuu 18.9 on Xubuntu 16.04

**Comment by Pipeliner** [comment]
The problem disappeared after upgrading to 18.04.

**Comment by CottonEaster** [comment]
> The problem disappeared after upgrading to Xubuntu 18.04.

I'm on Mint 19, which is based on Ubuntu 18.04, and I still have the problem.

**Comment by ams001** [comment]
On 17 November 2018 15:59:19 GNU_Linux_User <notifications@github.com> wrote:
> The problem disappeared after upgrading to Xubuntu 18.04.
>
> I'm on Mint 19, which is based on Ubuntu 18.04, and I still have the problem.

Likewise.

**Comment by CottonEaster** [comment]
@ dukisvk

Thank you. However, I have trouble understanding what you posted. Perhaps you could format the commands in a nicer (clearer) way?

**Comment by gothicVI** [comment]
@CottonEaster I think @dukisvk meant:
`cd /usr/lib/x86_64-linux-gnu/`
`sudo rm libcurl.so.4`
`sudo ln -s libcurl.so.4.5.0 libcurl.so.4`

**Comment by CottonEaster** [comment]
@gothicVI 

Thanks. I take it that the commands make version 4 of libcurl point to version 4.5 of libcurl. I wonder whether doing that might cause problems for other programs.

**Comment by dukisvk** [comment]
@gothicVI you can see where your symlink are pointing now with command: **ls -al | grep libcurl**
![image](https://user-images.githubusercontent.com/46889263/51476710-9be7e580-1d86-11e9-8e4e-7f8e9c2f4f35.png)

**Comment by dukisvk** [comment]
@CottonEaster i had this problem because of this issue as i wrote above. There are some applications which need to run with libcurl version 3 and others need libcurl version 4. This is problem of this library, which is unfortunatelly not back-compatible. I have to do this workaround for several apps. But as I said, for this purpose there is LD_LIBRARY_PATH command/variable, which solve this issue. By using this variable, you can force the applications to not use the default library folder but some otherone, where you have the proper versions of libraries.  
But maybe there is some other issue as well. If you run **ukuu --version**, you will see if you have some error messages or not in output. 
But i have upgraded kernel just now and had the feeling, that it works now without problem. (4.20.3-042003-generic)

**Comment by gothicVI** [comment]
> I wonder whether doing that might cause problems for other programs.
That's also what I'm not completely sure about.

> @gothicVI you can see where your symlinks are pointing now with command: **ls -al | grep libcurl**
I'm aware of that ;)
Anyhow, please try to stick to the github syntax, i.e. provide code in a  `code environment` and not in **bolt font** or via image. It makes it almost impossible to quickly follow your comments.
